duckdb/01_getting_started.py

```diff
@@ -0,0 +1,1652 @@
+
+
+    # Distribution of account balances
+    balance_data = new_memory_db.execute(
+        """
+        SELECT 
+            name,
+            account_balance
+        FROM users_memory
+        ORDER BY account_balance DESC
+        """
+    ).df()
+
+    fig4 = plotly_express.pie(
+        balance_data,
+        names="name",
+        values="account_balance",
+        title="Distribution of Account Balances",
+        labels={"account_balance": "Account Balance ($)", "name": "User"},
+        color_discrete_sequence=plotly_express.colors.qualitative.Pastel,
+    )
+    fig4.update_traces(textinfo="percent+label", textposition="inside")
+    fig4.update_layout(height=450, margin=dict(t=50, b=50))
+
+
+    category_tabs = mo.ui.tabs(
+        {
+            "Age Group Analysis": mo.vstack(
+                [
+                    mo.ui.tabs(
+                        {
+                            "User Distribution": mo.ui.plotly(fig1),
+                            "Average Balance": mo.ui.plotly(fig2),
+                        }
+                    )
+                ]
+            ),
+            "Financial Analysis": mo.vstack(
+                [
+                    mo.ui.tabs(
+                        {
+                            "Age vs Balance": mo.ui.plotly(fig3),
+                            "Balance Distribution": mo.ui.plotly(fig4),
+                        }
+                    )
+                ]
+            ),
+        },
+        lazy=True,
+    )
+
+    mo.vstack(
+        [
+            mo.md("### Select a visualization category:"),
+            category_tabs,
+        ],
+        gap=1.5,
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""
+    # [9. Database Management Best Practices]
+
+    ### Closing Connections
+
+    It's important to close database connections when you're done with them, especially for file-based connections:
+
+    ```python
+    memory_db.close()
+    file_db.close()
+    ```
+
+    ### Transaction Management
+
+    DuckDB supports transactions, which can be useful for more complex operations:
+
+    ```python
+    conn = duckdb.connect('mydb.db')
+    conn.begin()  # Start transaction
+
+    try:
+        conn.execute("INSERT INTO users VALUES (1, 'Test User')")
+        conn.execute("UPDATE balances SET amount = amount - 100 WHERE user_id = 1")
+        conn.commit()  # Commit changes
+    except:
+        conn.rollback()  # Undo changes if error
+        raise
+    ```
+
+    ### Query Performance
+
+    DuckDB is optimized for analytical queries. For best performance:
+
+    - Use appropriate data types
+    - Create indexes for frequently queried columns
+    - For large datasets, consider partitioning
+    - Use prepared statements for repeated queries
+    """
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _interactive_dashboard(mo):
+    mo.md(rf"""## 10. Interactive DuckDB Dashboard with Marimo and Plotly""")
```

Comment:
```suggestion
    mo.md(rf"""## 10. Interactive DuckDB Dashboard with marimo and Plotly""")
```
We generally prefer lowercase "m". Would be nice to see the changes throughout the notebook.